Lib/test/test_httplib.py

```diff
@@ -1877,6 +1877,16 @@ def test_connect_with_tunnel(self):
         # This test should be removed when CONNECT gets the HTTP/1.1 blessing
         self.assertNotIn(b'Host: proxy.com', self.conn.sock.data)
 
+    def test_connect_with_tunnel_idna(self):
+        dest = '\u03b4\u03c0\u03b8.gr'
+        expected = 'CONNECT %s:%d HTTP/1.1\r\n'.encode('ascii') % (
```

Comment:
Is there a reason to not use `b'CONNECT %s:%d HTTP/1.1\r\n'` instead? We already know that "HTTP/1.1", `dest.encode('idna')` and `client.HTTP_PORT` don't contain any non-ASCII chars so there is no need to encode it ASCII.